pulp/pulp.py

```diff
@@ -1063,7 +1063,7 @@ def __init__(self, e=None, sense=const.LpConstraintEQ, name=None, rhs=None):
         """
         self.expr = e if isinstance(e, LpAffineExpression) else LpAffineExpression(e)
         self.name = name
-        self.constant: float = self.expr.constant
```

Comment:
This just hides a potential bug. Better to fix the annotations

https://github.com/coin-or/pulp/pull/807/files#diff-b2a258edfe0fa294f032a978656a4082842558faa98b0950d9f68c50f546bbe2R169

https://github.com/coin-or/pulp/pull/807/files#diff-b2a258edfe0fa294f032a978656a4082842558faa98b0950d9f68c50f546bbe2R1212

Reply:
In my PR, I added some type aliases, which I then used where appropriate.
```python
LptNumber: TypeAlias = int | float
LptItem: TypeAlias = Union[LptNumber, "LpVariable"]
LptExpr: TypeAlias = Union[LptItem, "LpAffineExpression"]
LptConstExpr: TypeAlias = Union[LptNumber, "LpAffineExpression"]
LptExprConstraint: TypeAlias = Union[LptExpr, "LpConstraint"]
LptVars: TypeAlias = "LpVariable"
```